src/session.ts

```diff
@@ -719,6 +726,16 @@ export class SessionManager implements Middleware {
             .showQuickPick<SessionMenuItem>(menuItems)
             .then((selectedItem) => { selectedItem.callback(); });
     }
+
+    private isLoginShell(pwshPath: string): boolean {
+        try {
+            cp.execFileSync(pwshPath, ["-Login", "-NoProfile", "-NoLogo", "-Command", "exit 0"]);
```

Comment:
What does this throw if it's not the login shell?

Also, did your change to allow PowerShell to be a login shell make it in PS 6.2?

Reply:
> What does this throw if it's not the login shell?

The [`child_process` docs](https://nodejs.org/api/child_process.html#child_process_child_process_execfilesync_file_args_options) say:

> If the process times out or has a non-zero exit code, this method will throw an Error that will include the full result of the underlying child_process.spawnSync().

So when the `-Login` switch is not recognised and PowerShell exits with a non-zero exit code, node throws, we catch and return `false`.

> Also, did your change to allow PowerShell to be a login shell make it in PS 6.2?

No, only in 7 I think. But since you can install PowerShell to any path and configure VSCode to pick that up, we have no way of knowing what version a PS executable is, so are forced to run it to check.

I'll add a comment.

I should also say I haven't tested this, since I've been working on Windows. Why it's still in draft.

Reply:
Sorry I meant that I think  `-Login` existed in 6.2 but it was a no-op so that might be a false positive.

Reply:
Ohhh, I see! We might need @SteveL-MSFT's thoughts there. I think there was indeed a `-l` in either 6.2 or a 7 preview, although it possibly was short for `-LoadProfile`

Reply:
@rjmholt is correct, 6.2 only supported `-l` as short for `-loadprofile` making it "compatible" with Unix expecting `-l` to be accepted.  `-login` will fail on 6.2

Reply:
This is in 6.2.2 @SteveL-MSFT . This is a weird error output, isn't it?
```
pwsh -l
Invalid argument '-l', did you mean:

  -nologo
  -noprofile
  -windowstyle
  -file
  -executionpolicy
  -settingsfile
  -help
  -removeworkingdirectorytrailingcharacter
```

Reply:
My memory is failing me.  Looks like it was added in 7.0 but has been superceded by Rob's change to support -login: https://github.com/PowerShell/PowerShell/pull/9528